sky/serve/controller.py

```diff
@@ -152,7 +152,10 @@ def configure_logger():
         logger.info('SkyServe Controller started on '
                     f'http://localhost:{self._port}')
 
-        uvicorn.run(self._app, host='localhost', port=self._port)
+        # We expose the controller to the public network to allow external
+        # load balancers (example, for high availability load balancers) to
+        # communicate with the controller.
+        uvicorn.run(self._app, host='0.0.0.0', port=self._port)
```

Comment:
Adding auth might out of scope for this PR. For now, I have changed the logic to expose on `0.0.0.0` only if running inside kubernetes: https://github.com/skypilot-org/skypilot/pull/3377/commits/803df4ab7dd95f9e7f3834199071c501e6207052

This will be accessible from only within the kubernetes cluster (which is a trusted domain) and is not exposed to the public internet. Does that work?